### ISSUE TITLE
didFinishInstallingUpdate is called multiple times

### ISSUE THREAD

**Issue by TheRusskiy** [stripe/stripe-terminal-react-native#361]
**Describe the bug**
didFinishInstallingUpdate is called multiple times

**To Reproduce**
```
useEffect(() => {
    const finishInstallingUpdateListener =
      stripeTerminalRef.current.emitter?.addListener(
        FINISH_INSTALLING_UPDATE,
        data => {
          console.log('didFinishInstallingUpdateListener')
          Alert.alert('Update Installed', 'Terminal update was installed', [
            {
              text: 'Ok'
            }
          ])
          setUpdateStatus('finished')
        }
      )

    return () => {
      finishInstallingUpdateListener?.remove()
    }
  }, [])
```

**Expected behavior**
The callback only called once

**Smartphone (please complete the following information):**

- Device: iPad simulator
- Version 0.0.1-beta.6

**Additional context**
```
 LOG  {"availableUpdate": null, "batteryLevel": null, "batteryStatus": "unknown", "deviceSoftwareVersion": null, "deviceType": "chipper2X", "id": null, "ipAddress": null, "isCharging": null, "label": null, "location": {"address": {"city": "South San Francisco", "country": "US", "line1": "354 Oyster Point Blvd", "line2": null, "postalCode": "94080", "state": "CA"}, "displayName": "Stripe HQ", "id": "st_simulated", "livemode": false}, "locationId": "st_simulated", "locationStatus": "set", "serialNumber": "CHB20SIMULATOR1", "simulated": true, "status": "offline"}
 LOG  [Stripe terminal]: didUpdateDiscoveredReaders [{"availableUpdate": null, "batteryLevel": null, "batteryStatus": "unknown", "deviceSoftwareVersion": null, "deviceType": "chipper2X", "id": null, "ipAddress": null, "isCharging": null, "label": null, "location": {"address": [Object], "displayName": "Stripe HQ", "id": "st_simulated", "livemode": false}, "locationId": "st_simulated", "locationStatus": "set", "serialNumber": "CHB20SIMULATOR1", "simulated": true, "status": "offline"}]
 LOG  [Stripe terminal]: didChangeConnectionStatus connecting
 LOG  Connection token: pst_test_xxxx
 LOG  [Stripe terminal]: didFinishInstallingUpdate {}
 LOG  [Stripe terminal]: didFinishInstallingUpdate {}
 LOG  [Stripe terminal]: didFinishInstallingUpdate {}
 LOG  [Stripe terminal]: didFinishInstallingUpdate {}
 LOG  [Stripe terminal]: didFinishInstallingUpdate {}
 LOG  [Stripe terminal]: didFinishInstallingUpdate {}
 LOG  [Stripe terminal]: didChangeConnectionStatus connected
 LOG  [Stripe terminal]: didChangePaymentStatus ready
 LOG  [Stripe terminal]: didReportAvailableUpdate {"deviceSoftwareVersion": "1.00.03.32-SZZZ_Generic_v37-300001", "estimatedUpdateTime": "estimate1To2Minutes", "requiredAt": "1657992968529"}
 LOG  [Stripe terminal]: didChangePaymentStatus notReady
 LOG  [Stripe terminal]: didStartInstallingUpdate {"deviceSoftwareVersion": "1.00.03.32-SZZZ_Generic_v37-300001", "estimatedUpdateTime": "estimate1To2Minutes", "requiredAt": "1657992968529"}
 LOG  didStartInstallingUpdateListener
 LOG  [Stripe terminal]: didReportReaderSoftwareUpdateProgress {"progress": "0.0"}
 LOG  [Stripe terminal]: didFinishDiscoveringReaders {"error": null}
 LOG  [Stripe terminal]: didReportReaderSoftwareUpdateProgress {"progress": "0.01"}
 LOG  [Stripe terminal]: didFinishInstallingUpdate {"deviceSoftwareVersion": "1.00.03.32-SZZZ_Generic_v37-300001", "error": {"error": {"code": "Busy", "message": "Could not execute installUpdate because the SDK is busy with another command: installUpdate."}}, "estimatedUpdateTime": "estimate1To2Minutes", "requiredAt": "1657992968529"}
 LOG  [Stripe terminal]: didReportReaderSoftwareUpdateProgress {"progress": "0.02"}
 LOG  [Stripe terminal]: didReportReaderSoftwareUpdateProgress {"progress": "0.03"}
 LOG  [Stripe terminal]: didReportReaderSoftwareUpdateProgress {"progress": "0.04"}
 LOG  [Stripe terminal]: didReportReaderSoftwareUpdateProgress {"progress": "0.05"}
 LOG  [Stripe terminal]: didReportReaderSoftwareUpdateProgress {"progress": "0.06"}
 LOG  [Stripe terminal]: didReportReaderSoftwareUpdateProgress {"progress": "0.07"}
 LOG  [Stripe terminal]: didReportReaderSoftwareUpdateProgress {"progress": "0.08"}
 LOG  [Stripe terminal]: didFinishInstallingUpdate {"deviceSoftwareVersion": "1.00.03.32-SZZZ_Generic_v37-300001", "error": {"error": {"code": "Busy", "message": "Could not execute installUpdate because the SDK is busy with another command: installUpdate."}}, "estimatedUpdateTime": "estimate1To2Minutes", "requiredAt": "1657992968529"}
 LOG  [Stripe terminal]: didReportReaderSoftwareUpdateProgress {"progress": "0.09"}
 LOG  [Stripe terminal]: didReportReaderSoftwareUpdateProgress {"progress": "0.1"}
 LOG  [Stripe terminal]: didReportReaderSoftwareUpdateProgress {"progress": "0.11"}
 LOG  [Stripe terminal]: didReportReaderSoftwareUpdateProgress {"progress": "0.12"}
 LOG  [Stripe terminal]: didReportReaderSoftwareUpdateProgress {"progress": "0.13"}
 LOG  [Stripe terminal]: didReportReaderSoftwareUpdateProgress {"progress": "0.14"}
 LOG  [Stripe terminal]: didReportReaderSoftwareUpdateProgress {"progress": "0.15"}
 LOG  [Stripe terminal]: didReportReaderSoftwareUpdateProgress {"progress": "0.16"}
 LOG  [Stripe terminal]: didReportReaderSoftwareUpdateProgress {"progress": "0.17"}
 LOG  [Stripe terminal]: didReportReaderSoftwareUpdateProgress {"progress": "0.18"}
 LOG  [Stripe terminal]: didReportReaderSoftwareUpdateProgress {"progress": "0.19"}
 LOG  [Stripe terminal]: didReportReaderSoftwareUpdateProgress {"progress": "0.2"}
 LOG  [Stripe terminal]: didReportReaderSoftwareUpdateProgress {"progress": "0.21"}
 LOG  [Stripe terminal]: didReportReaderSoftwareUpdateProgress {"progress": "0.22"}
 LOG  [Stripe terminal]: didReportReaderSoftwareUpdateProgress {"progress": "0.23"}
 LOG  [Stripe terminal]: didReportReaderSoftwareUpdateProgress {"progress": "0.24"}
 LOG  [Stripe terminal]: didReportReaderSoftwareUpdateProgress {"progress": "0.25"}
 LOG  [Stripe terminal]: didReportReaderSoftwareUpdateProgress {"progress": "0.26"}
 LOG  [Stripe terminal]: didReportReaderSoftwareUpdateProgress {"progress": "0.27"}
 LOG  [Stripe terminal]: didReportReaderSoftwareUpdateProgress {"progress": "0.28"}
 LOG  [Stripe terminal]: didReportReaderSoftwareUpdateProgress {"progress": "0.29"}
 LOG  [Stripe terminal]: didReportReaderSoftwareUpdateProgress {"progress": "0.3"}
 LOG  [Stripe terminal]: didReportReaderSoftwareUpdateProgress {"progress": "0.31"}
 LOG  [Stripe terminal]: didReportReaderSoftwareUpdateProgress {"progress": "0.32"}
 LOG  [Stripe terminal]: didReportReaderSoftwareUpdateProgress {"progress": "0.33"}
 LOG  [Stripe terminal]: didReportReaderSoftwareUpdateProgress {"progress": "0.34"}
 LOG  [Stripe terminal]: didReportReaderSoftwareUpdateProgress {"progress": "0.35"}
 LOG  [Stripe terminal]: didReportReaderSoftwareUpdateProgress {"progress": "0.36"}
 LOG  [Stripe terminal]: didReportReaderSoftwareUpdateProgress {"progress": "0.37"}
 LOG  [Stripe terminal]: didReportReaderSoftwareUpdateProgress {"progress": "0.38"}
 LOG  [Stripe terminal]: didReportReaderSoftwareUpdateProgress {"progress": "0.39"}
 LOG  [Stripe terminal]: didReportReaderSoftwareUpdateProgress {"progress": "0.4"}
 LOG  [Stripe terminal]: didReportReaderSoftwareUpdateProgress {"progress": "0.41"}
 LOG  [Stripe terminal]: didReportReaderSoftwareUpdateProgress {"progress": "0.42"}
 LOG  [Stripe terminal]: didReportReaderSoftwareUpdateProgress {"progress": "0.43"}
 LOG  [Stripe terminal]: didReportReaderSoftwareUpdateProgress {"progress": "0.44"}
 LOG  [Stripe terminal]: didReportReaderSoftwareUpdateProgress {"progress": "0.45"}
 LOG  [Stripe terminal]: didReportReaderSoftwareUpdateProgress {"progress": "0.46"}
 LOG  [Stripe terminal]: didReportReaderSoftwareUpdateProgress {"progress": "0.47"}
 LOG  [Stripe terminal]: didReportReaderSoftwareUpdateProgress {"progress": "0.48"}
 LOG  [Stripe terminal]: didReportReaderSoftwareUpdateProgress {"progress": "0.49"}
 LOG  [Stripe terminal]: didReportReaderSoftwareUpdateProgress {"progress": "0.5"}
 LOG  [Stripe terminal]: didReportReaderSoftwareUpdateProgress {"progress": "0.51"}
 LOG  [Stripe terminal]: didReportReaderSoftwareUpdateProgress {"progress": "0.52"}
 LOG  [Stripe terminal]: didReportReaderSoftwareUpdateProgress {"progress": "0.53"}
 LOG  [Stripe terminal]: didReportReaderSoftwareUpdateProgress {"progress": "0.54"}
 LOG  [Stripe terminal]: didReportReaderSoftwareUpdateProgress {"progress": "0.55"}
 LOG  [Stripe terminal]: didReportReaderSoftwareUpdateProgress {"progress": "0.56"}
 LOG  [Stripe terminal]: didReportReaderSoftwareUpdateProgress {"progress": "0.57"}
 LOG  [Stripe terminal]: didReportReaderSoftwareUpdateProgress {"progress": "0.58"}
 LOG  [Stripe terminal]: didReportReaderSoftwareUpdateProgress {"progress": "0.59"}
 LOG  [Stripe terminal]: didReportReaderSoftwareUpdateProgress {"progress": "0.6"}
 LOG  [Stripe terminal]: didReportReaderSoftwareUpdateProgress {"progress": "0.61"}
 LOG  [Stripe terminal]: didReportReaderSoftwareUpdateProgress {"progress": "0.62"}
 LOG  [Stripe terminal]: didReportReaderSoftwareUpdateProgress {"progress": "0.63"}
 LOG  [Stripe terminal]: didReportReaderSoftwareUpdateProgress {"progress": "0.64"}
 LOG  [Stripe terminal]: didReportReaderSoftwareUpdateProgress {"progress": "0.65"}
 LOG  [Stripe terminal]: didReportReaderSoftwareUpdateProgress {"progress": "0.66"}
 LOG  [Stripe terminal]: didReportReaderSoftwareUpdateProgress {"progress": "0.67"}
 LOG  [Stripe terminal]: didReportReaderSoftwareUpdateProgress {"progress": "0.68"}
 LOG  [Stripe terminal]: didReportReaderSoftwareUpdateProgress {"progress": "0.69"}
 LOG  [Stripe terminal]: didReportReaderSoftwareUpdateProgress {"progress": "0.7"}
 LOG  [Stripe terminal]: didReportReaderSoftwareUpdateProgress {"progress": "0.71"}
 LOG  [Stripe terminal]: didReportReaderSoftwareUpdateProgress {"progress": "0.72"}
 LOG  [Stripe terminal]: didReportReaderSoftwareUpdateProgress {"progress": "0.73"}
 LOG  [Stripe terminal]: didReportReaderSoftwareUpdateProgress {"progress": "0.74"}
 LOG  [Stripe terminal]: didReportReaderSoftwareUpdateProgress {"progress": "0.75"}
 LOG  [Stripe terminal]: didReportReaderSoftwareUpdateProgress {"progress": "0.76"}
 LOG  [Stripe terminal]: didReportReaderSoftwareUpdateProgress {"progress": "0.77"}
 LOG  [Stripe terminal]: didReportReaderSoftwareUpdateProgress {"progress": "0.78"}
 LOG  [Stripe terminal]: didReportReaderSoftwareUpdateProgress {"progress": "0.79"}
 LOG  [Stripe terminal]: didReportReaderSoftwareUpdateProgress {"progress": "0.8"}
 LOG  [Stripe terminal]: didReportReaderSoftwareUpdateProgress {"progress": "0.81"}
 LOG  [Stripe terminal]: didReportReaderSoftwareUpdateProgress {"progress": "0.82"}
 LOG  [Stripe terminal]: didReportReaderSoftwareUpdateProgress {"progress": "0.83"}
 LOG  [Stripe terminal]: didReportReaderSoftwareUpdateProgress {"progress": "0.84"}
 LOG  [Stripe terminal]: didReportReaderSoftwareUpdateProgress {"progress": "0.85"}
 LOG  [Stripe terminal]: didReportReaderSoftwareUpdateProgress {"progress": "0.86"}
 LOG  [Stripe terminal]: didReportReaderSoftwareUpdateProgress {"progress": "0.87"}
 LOG  [Stripe terminal]: didReportReaderSoftwareUpdateProgress {"progress": "0.88"}
 LOG  [Stripe terminal]: didReportReaderSoftwareUpdateProgress {"progress": "0.89"}
 LOG  [Stripe terminal]: didReportReaderSoftwareUpdateProgress {"progress": "0.9"}
 LOG  [Stripe terminal]: didReportReaderSoftwareUpdateProgress {"progress": "0.91"}
 LOG  [Stripe terminal]: didReportReaderSoftwareUpdateProgress {"progress": "0.92"}
 LOG  [Stripe terminal]: didReportReaderSoftwareUpdateProgress {"progress": "0.93"}
 LOG  [Stripe terminal]: didReportReaderSoftwareUpdateProgress {"progress": "0.94"}
 LOG  [Stripe terminal]: didReportReaderSoftwareUpdateProgress {"progress": "0.95"}
 LOG  [Stripe terminal]: didReportReaderSoftwareUpdateProgress {"progress": "0.96"}
 LOG  [Stripe terminal]: didReportReaderSoftwareUpdateProgress {"progress": "0.97"}
 LOG  [Stripe terminal]: didReportReaderSoftwareUpdateProgress {"progress": "0.98"}
 LOG  [Stripe terminal]: didReportReaderSoftwareUpdateProgress {"progress": "0.99"}
 LOG  [Stripe terminal]: didChangePaymentStatus ready
 LOG  [Stripe terminal]: didFinishInstallingUpdate {"deviceSoftwareVersion": "1.00.03.32-SZZZ_Generic_v37-300001", "estimatedUpdateTime": "estimate1To2Minutes", "requiredAt": "1657992968529"}
```

**Comment by jdivock-stripe** [comment]
Hrm, I'm having trouble getting a repro, attaching a debug statement here https://github.com/stripe/stripe-terminal-react-native/blob/a0af13559e6948d5319e787028d1f86f9292d3f4/src/components/StripeTerminalProvider.tsx#L142 only shows the event getting emitted once.

- Can you try on beta.7?
- I know `useEffect` can get out of hand with it's dependencies, can you ensure only one listener is being mounted?
- Do you see this behavior if you make use of the  `onDidFinishInstallingUpdate` callback returned from `useStripeTerminal`?

Output from our reference implementation shows the following:

```
 LOG  [Stripe terminal]: didChangeConnectionStatus connecting
 LOG  [Stripe terminal]: didStartInstallingUpdate {"deviceSoftwareVersion": "2.01.00.08-SZZZ_US_v9-300001", "estimatedUpdateTime": "estimate1To2Minutes", "requiredAt": "1657645234598"}
 LOG  [Stripe terminal]: didReportReaderSoftwareUpdateProgress {"progress": "0.0"}
 LOG  [Stripe terminal]: didReportReaderSoftwareUpdateProgress {"progress": "0.01"}
 
 ...
 LOG  [Stripe terminal]: didReportReaderSoftwareUpdateProgress {"progress": "0.98"}
 LOG  [Stripe terminal]: didReportReaderSoftwareUpdateProgress {"progress": "0.99"}
 LOG  [Stripe terminal]: didFinishInstallingUpdate {"deviceSoftwareVersion": "2.01.00.08-SZZZ_US_v9-300001", "estimatedUpdateTime": "estimate1To2Minutes", "requiredAt": "1657645234598"}
 LOG  [Stripe terminal]: didChangeConnectionStatus connected
 LOG  [Stripe terminal]: didChangePaymentStatus ready
```

**Comment by TheRusskiy** [comment]
@jdivock-stripe thanks, I think something might have fixed it in 0.7 version (at least it didn't happen to me the last time after upgrading). I am going to keep an eye on this and if this issue re-appears I am going to let you know.
